### ISSUE TITLE
osx layer can no longer find gls

### ISSUE THREAD

**Issue by jeffhertzler** [syl20bnr/spacemacs#10957]
#### Description :octocat:
osx layer can no longer find `gls`

Relevant `spacemacs.env` entry:
`PATH=/Users/jhertzler/bin:/Users/jhertzler/.asdf/bin:/Users/jhertzler/.asdf/shims:/Users/jhertzler/.config/fnm/bin:/Applications/Emacs.app/Contents/MacOS/bin-x86_64-10_10:/Applications/Emacs.app/Contents/MacOS/libexec-x86_64-10_10:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin`

#### Reproduction guide :beetle:
- Start Emacs
- `SPC h d v` `insert-directory-program`

*Observed behaviour:* :eyes: :broken_heart:
value is `ls`

*Expected behaviour:* :heart: :smile:
value is `/usr/local/bin/gls`

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 6220ace29)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-snippets-in-popup t)
 colors crystal csv elixir elm emacs-lisp java
 (javascript :variables tern-disable-port-files nil)
 git github go html ivy lua
 (markdown :variables markdown-live-preview-engine 'vmd)
 nginx osx php prettier-js react restclient ruby shell shell-scripts syntax-checking tern typescript version-control vimscript vinegar yaml)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS

**Comment by jeffhertzler** [comment]
You can add
```
(let ((gls (executable-find "gls")))
 (when gls
   (setq insert-directory-program gls
         dired-listing-switches "-aBhl --group-directories-first")))
```

to your `dotspacemacs/user-config` for now to get the functionality back

**Comment by syl20bnr** [comment]
I think the PATH order should be reversed. I'll fix it tomorrow while I
finish the work on the environment variables support.

**Comment by syl20bnr** [comment]
Should be fixed with last develop with the additional work on env vars management, see commit https://github.com/syl20bnr/spacemacs/commit/a013d8687 for more info.

**Comment by bcc32** [comment]
I believe this issue should now be fixed by 17c9906 in the develop branch.
